cmd/prometheus/main.go

```diff
@@ -1173,7 +1314,14 @@ func (s *readyStorage) ChunkQuerier(ctx context.Context, mint, maxt int64) (stor
 
 func (s *readyStorage) ExemplarQuerier(ctx context.Context) (storage.ExemplarQuerier, error) {
 	if x := s.get(); x != nil {
-		return x.ExemplarQuerier(ctx)
+		switch db := x.(type) {
```

Comment:
Is this switch necessary? Wouldn't `db.ExemplarQuerier(ctx)` just end up returning a `wal.ErrUnsupported` anyway?

Reply:
It's only necessary because `s.db` is `storage.Storage` and doesn't have the `storage.ExemplarQueryable` interface